httpclient-core/src/main/java/esa/httpclient/core/CompositeRequest.java

```diff
@@ -0,0 +1,336 @@
+        return client.execute(this, ctx, handle, handler);
+    }
+
+    @Override
+    public synchronized PlainRequest body(byte[] bytes) {
+        checkStarted();
+        cleanBody();
+        this.type = 0;
+        this.bytes = bytes;
+        return self();
+    }
+
+    @Override
+    public MultipartRequest multipart() {
+        this.type = 1;
+        return this;
+    }
+
+    @Override
+    public ChunkRequest segment() {
+        this.type = 2;
+        return request.get();
+    }
+
+    @Override
+    public synchronized FileRequest body(File file) {
+        checkStarted();
+        cleanBody();
+        this.type = 3;
+        this.file = file;
+        return self();
+    }
+
+    @Override
+    public byte[] bytes() {
+        return bytes;
+    }
+
+    @Override
+    public File file() {
+        return file;
+    }
+
+    @Override
+    public synchronized MultipartRequest multipartEncode(boolean multipartEncode) {
+        checkStarted();
+        this.multipartEncode = multipartEncode;
+        return self();
+    }
+
+    @Override
+    public boolean multipartEncode() {
+        return multipartEncode;
+    }
+
+    @Override
+    public synchronized MultipartRequest attribute(String name, String value) {
+        checkStarted();
+        if (illegalArgs(name, value)) {
+            return self();
+        }
+        attributes.add(name, value);
+        return self();
+    }
+
+    @Override
+    public synchronized MultipartRequest file(String name, File file) {
+        checkStarted();
+        return file(name, file, DEFAULT_BINARY_CONTENT_TYPE);
+    }
+
+    @Override
+    public synchronized MultipartRequest file(String name, File file, String contentType) {
+        checkStarted();
+        return file(name, file, contentType, DEFAULT_TEXT_CONTENT_TYPE.equalsIgnoreCase(contentType));
+    }
+
+    @Override
+    public synchronized MultipartRequest file(String name,
+                                              File file,
+                                              String contentType,
+                                              boolean isText) {
+        checkStarted();
+        if (illegalArgs(name, file)) {
+            return self();
+        }
+        checkMultipartFile();
+        return file(name, file.getName(), file, contentType, isText);
+    }
+
+    @Override
+    public synchronized MultipartRequest file(String name,
+                                              String filename,
+                                              File file,
+                                              String contentType,
+                                              boolean isText) {
+        checkStarted();
+        checkMultipartFile();
+        files.add(new MultipartFileItem(name, filename, file, contentType, isText));
+        return self();
+    }
+
+    @Override
+    public MultiValueMap<String, String> attributes() {
+        return new HashMultiValueMap<>(attributes);
+    }
```

Comment:
duplicated `synchronized ` and check in overload methods

Reply:
done